File: Plugins/Aspose_Tasks_Java_for_Ruby/lib/asposetasksjava/Projects/saveprojectdatatootherformats.rb
module Asposetasksjava
  module SaveProjectDataToOtherFormats
    def initialize()
        # Saving a Project as CSV
        save_to_csv()

        # Saving a Projects as Text
        save_to_text()
    end

    def save_to_csv()
        data_dir = File.dirname(File.dirname(File.dirname(File.dirname(__FILE__)))) + '/data/'
                
        # Instantiate project object
        project = Rjb::import('com.aspose.tasks.Project').new(data_dir + 'test_tasks.mpp')

        # Save the Project as CSV
        project.save(data_dir + "Project.csv", Rjb::import('com.aspose.tasks.SaveFileFormat').CSV)

        puts "Saved project data to CSV, please check the output file"
    end

    def save_to_text()
        data_dir = File.dirname(File.dirname(File.dirname(File.dirname(__FILE__)))) + '/data/'
                
        # Instantiate project object
        project = Rjb::import('com.aspose.tasks.Project').new(data_dir + 'test_tasks.mpp')

        # Save the Project as Text
        project.save(data_dir + "Project.txt", Rjb::import('com.aspose.tasks.SaveFileFormat').TXT)

        puts "Saved project data to Text, please check the output file"
    end
  end
end

File: Plugins/Aspose_Tasks_Java_for_PHP/src/aspose/tasks/WorkingWithProjects/SaveProjectDataToOtherFormats.php
<?php
namespace Aspose\Tasks\WorkingWithProjects;

use com\aspose\tasks\Project as Project;
use com\aspose\tasks\SaveFileFormat as SaveFileFormat;

class SaveProjectDataToOtherFormats{

    public static function run($dataDir=null){

        # Saving a Project as CSV
        SaveProjectDataToOtherFormats::save_to_csv($dataDir);

        # Saving a Projects as Text
        SaveProjectDataToOtherFormats::save_to_text();
    }

    public static function save_to_csv($dataDir=null){

        # Instantiate project object
        $project = new Project($dataDir . 'test_tasks.mpp');

        # Save the Project as CSV
        $saveFileFormat=new SaveFileFormat();
        $project->save($dataDir . "Project.csv", $saveFileFormat->CSV);
        print "Saved project data to CSV, please check the output file".PHP_EOL;
    }

    public static function save_to_text($dataDir=null){

        # Instantiate project object
        $project = new Project($dataDir . 'test_tasks.mpp');

        # Save the Project as Text
        $saveFileFormat=new SaveFileFormat();
        $project->save($dataDir . "Project.txt", $saveFileFormat->TXT);
        print "Saved project data to Text, please check the output file".PHP_EOL;
    }

}
?>